core/dbt/contracts/files.py

```diff
@@ -259,6 +260,28 @@ def get_tests(self, yaml_key, name):
                 return self.data_tests[yaml_key][name]
         return []
 
+    def add_metrics_from_measures(self, semantic_model_name: str, metric_unique_id: str):
+        if self.generated_metrics:
+            self.fix_metrics_from_measures()
+        if semantic_model_name not in self.metrics_from_measures:
+            self.metrics_from_measures[semantic_model_name] = []
+        self.metrics_from_measures[semantic_model_name].append(metric_unique_id)
+
+    def fix_metrics_from_measures(self):
+        # Loop through yaml dictionary looking for matching generated metrics
+        generated_metrics = self.generated_metrics
+        self.generated_metrics = []
```

Comment:
Are we using this manipulation of generated_metrics to avoid `fix_metrics_from_measures` and `fix_metrics_from_measures` infinite looping? Can we try to organize the function so it is easier to read?